src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -383,22 +351,32 @@ def __init__(self, target: str, options: ChannelArgumentType,
         # No new calls will be accepted by the Cython channel.
         self._channel.closing()
 
-        if grace:
-            # pylint: disable=unused-variable
-            _, pending = await asyncio.wait(self._ongoing_calls.calls,
-                                            timeout=grace,
-                                            loop=self._loop)
-
-            if not pending:
-                return
-
-        # A new set is created acting as a shallow copy because
-        # when cancellation happens the calls are automatically
-        # removed from the originally set.
-        calls = WeakSet(data=self._ongoing_calls.calls)
+        # Iterate through running tasks
+        tasks = _all_tasks()
+        calls = []
+        call_tasks = []
+        for task in tasks:
+            stack = task.get_stack(limit=1)
+            if not stack:
```

Comment:
Can you add a note explaining when this will happen and why it's okay to `continue`?

Reply:
The exception case is that if the Task is created in an environment that doesn't have normal Python stack, like C-extensions. Here is an example:

```
<Task pending coro=<AioServer._server_main_loop()> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x10dbb9dc8>()] created at /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py:396> cb=[AioServer._serving_task_crash_handler()] created at /private/var/tmp/_bazel_lidiz/ee2c6ba1011b403418e3cd5ac063655b/execroot/com_github_grpc_grpc/bazel-out/darwin-fastbuild/bin/src/python/grpcio_tests/tests_aio/unit/close_channel_test.runfiles/com_github_grpc_grpc/src/python/grpcio/grpc/experimental/aio/_server.py:98>
```

Comment added:
```Python
# If the Task is created by a C-extension, the stack will be empty.
```